pkg/api/v1beta3/types.go

```diff
@@ -509,6 +509,33 @@ type EnvVar struct {
 	Name string `json:"name" description:"name of the environment variable; must be a C_IDENTIFIER"`
 	// Optional: defaults to "".
 	Value string `json:"value,omitempty" description:"value of the environment variable; defaults to empty string"`
+	// Optional: specify a source for the value of the EnvVar.
+	*EnvVarSource `json:"source,omitempty" description:"source for the environment variable's value; cannot be used if value is set"`
```

Comment:
What do we want the JSON to look like?

```
{ 
    "name": "myvar",
    "source": { "objectInfo": { "fieldPath": ".metadata.name" } }
}
```

or

```
{ 
    "name": "myvar",
    "objectInfo": { "fieldPath": ".metadata.name" }
}
```

or 

```
{ 
    "name": "myvar",
    "value": { "objectInfo": {"fieldPath": ".metadata.name" }}
}
```

If I parse this correctly it's #2 that we are getting, yes?